### PR TITLE
Feature/select proxy

### LINKED ISSUE #1869: Allow API owner to select one from several configured proxies
If there is more than one API Umbrella proxy defined for deployment, allow the API owner to select one of them in the Proxy backend settings.

# Definition of done
* [x] If there is more than one API Umbrella defined, owner can select any of the defined proxies, when adding or editing proxy backend settings.
* [x] If there is only one API Umbrella defined, a simple solution is available for defaulting to the only API Umbrella.
* [x] Proxy base path will come from selected proxy.
* [x] Show an alert to owner, if switching between proxies: "This change will potentially break API calls from applications, if API users are not warned about the change."

Open issues: If owner changes the proxy, do we need to delete the proxy backend from the original proxy? Users will have a broken link, is it our responsibility to notify or is it owner's responsibility to warn the users?

# Wireframes

* In API profile Proxy tab, a dropmenu would appear depending on if there are more than 1 proxy available for selection. 
Open question: in case of only one proxy avialable, should we show the name as hard coded? 
apiprofilemultiproxyselector

* On selecting a specific proxy, associated fields for that proxy would be populated. In the case for only one available proxy, the fields can appear instantly Proxy tab is selected for viewing:

apiprofilemultiproxyselector2

apiprofilemultiproxyselector3

* When API owner tries to switch from one proxy to another, the following alert window should appear:
apiprofilemultiproxyselector4

### PR DESCRIPTION
**Pull request by marla-singer**
closes #1869 

Knew bugs:
1. When add proxy at first and proxy base path isn't unique, after clicking on "Save & Publish" form has hidden
_(Suggestion: create another issue for fixing it)_
2. When change proxy the second or next times, PB config isn't deleted on previously proxy. PB config also isn't deleted via button Delete or via selected the first item "Select one proxy"
_(Suggestion: create another issue for fixing it)_

### For reviewer
Please test the next conditions:

1. For one proxy
 -- Insert/Update/Delete proxy backend information
2. For many proxies
 -- Insert/Update/Delete proxy backend information
3. Change proxy via dropdown list
4.  Add/Remove proxy and watch for proxy backend form

### CODE REVIEW

**Comment by marla-singer** [comment]
@apinf/developers Ready for review

File: proxy_backends/client/form/form.js [hunk 1]
```diff
@@ -1,162 +1,215 @@
...
+
+  const proxyCount = Proxies.find().count();
+
+  // If proxy is only one then set these id on default
+  if (proxyCount === 1) {
+    currentProxyId = Proxies.findOne()._id;
+  }
+
+  // If proxy id already exists then set these id
+  const proxyBackanedExists = instance.data.proxyBackend;
```
Comment (jykae) [inline proxy_backends/client/form/form.js]: fix variable name

**Comment by jykae** [comment]
@marla-singer basic functionality works, I am trying to look if I could help with known bugs.

**Comment by jykae** [comment]
Also use ESLint to fix some lint errors on files you have worked.

File: proxy_backends/client/methods/convert_to_apiU_object.js [hunk 2]
```diff
@@ -0,0 +1,23 @@
+// When there are updating the proxy backend information and changing proxy
```
Comment (jykae) [inline proxy_backends/client/methods/convert_to_apiU_object.js]: No need to save characters on file name, replace "apiU" -> "apiUmbrella

Comment (marla-singer) [inline proxy_backends/client/methods/convert_to_apiU_object.js]: Okay, done

File: proxy_backends/client/form/form.js [hunk 3]
```diff
@@ -1,162 +1,224 @@
...
   'click #delete-proxy-button': () => {
     /* Function procedure in generic form
     1. Delete API Backend on proxy (eg. API Umbrella)
       - call necessary functions by proxy type
     2. Delete Proxy Backend on Apinf
     */
 
+    // Notify users about deleting proxy
+    const message = TAPi18n.__('proxyBackend_confirmText_deleteProxyBackendInformation');
+    const confirmation = confirm(message);
```
Comment (jykae) [inline proxy_backends/client/form/form.js]: @marla-singer We have used nicer looking modal confirmations, check for example API delete https://github.com/apinf/platform/tree/develop/apis/client/delete

Documentation: https://github.com/PeppeL-G/bootstrap-3-modal

Comment (marla-singer) [inline proxy_backends/client/form/form.js]: How I can detecte that user click OK or Cancel button from 'click #delete-proxy-button' handler?

**Comment by marla-singer** [comment]
@jykae I replaced one of standart confirm dialog.
Is it good enogh?

**Comment by jykae** [comment]
@marla-singer I would not want to merge this before known bugs are solved, however PR should also not grow too large. I bring this up in daily, if we can leave something for continuance task.
Good work so far 👍

**Comment by jykae** [comment]
@marla-singer Found undefined template:
nayttokuva 2016-11-17 kello 10 00 15
nayttokuva 2016-11-17 kello 10 00 35

**Comment by marla-singer** [comment]
@jykae I suggest to replace standard confirm dialog box to bootstrap and test it again. We can create issue to knew and future bugs and point to this PR

Yes, I forgot to add template to git hub 😕

**Comment by marla-singer** [comment]
I will write you when implemented bootstrap modal form

**Comment by brylie** [comment]
@marla-singer, when testing the current revision, I am getting the following console error on the Proxy settings page:

```js
TypeError: Template.removeSelectedProxy is undefined
```

**Comment by brylie** [comment]
In Chrome (Chromium), I am unable to navigate to the Proxy tab for a given API backend. I cannot create a new API backend either. In both cases, the platform is throwing a console error:

```js
Exception in template helper: TypeError: Cannot read property 'namedContext' of undefined
    at Object.autoFormFieldMessage
```

**Comment by marla-singer** [comment]
@jykae @brylie You can try right now

**Comment by brylie** [comment]
Overall, this works and looks good. What are the next step(s)?

**Comment by marla-singer** [comment]
Let's merge it and I will create the issues for bugs from description

**Comment by brylie** [comment]
Sounds good.

**Pull request merged**